backend/alert/alerts.py

```diff
@@ -48,46 +49,134 @@ def __init__(self, template, reg):
             }
         )
+        self.close_text = None
+        if close_template:
+            t = loader.get_template(close_template)
+            self.close_text = t.render(
+                {
+                    "course": reg.section.full_code,
+                    "brand": "Penn Course Alert",
+                    "auto_resubscribe": reg.auto_resubscribe,
+                }
+            )
         self.registration = reg
 
     @abstractmethod
-    def send_alert(self):
+    def send_alert(self, close_notification=False):
         pass
 
 
 class Email(Alert):
     def __init__(self, reg):
-        super().__init__("alert/email_alert.html", reg)
+        super().__init__(
+            "alert/email_alert.html", reg, close_template="alert/email_close_alert.html"
+        )
 
-    def send_alert(self):
+    def send_alert(self, close_notification=False):
+        """
+        Returns False if notification was not sent intentionally,
+        and None if notification was attempted to be sent but an error occurred.
+        """
         if self.registration.user is not None and self.registration.user.profile.email is not None:
             email = self.registration.user.profile.email
         elif self.registration.email is not None:
             email = self.registration.email
         else:
             return False
+
         try:
+            if close_notification:
+                if not self.close_text:
+                    # This should be unreachable
+                    return None
+                alert_subject = "RE: %s is now open!" % self.registration.section.full_code
+                alert_text = self.close_text
+            else:
+                alert_subject = "%s is now open!" % self.registration.section.full_code
+                alert_text = self.text
             return send_email(
                 from_="Penn Course Alert <team@penncoursealert.com>",
                 to=email,
-                subject="%s is now open!" % self.registration.section.full_code,
-                html=self.text,
+                subject=alert_subject,
+                html=alert_text,
             )
         except SMTPRecipientsRefused:
             logger.exception("Email Error")
-            return False
+            return None
 
 
 class Text(Alert):
     def __init__(self, reg):
-        super().__init__("alert/text_alert.txt", reg)
-
-    def send_alert(self):
+        super().__init__("alert/text_alert.txt", reg, close_template="alert/text_alert_close.txt")
+
+    def send_alert(self, close_notification=False):
+        """
+        Returns False if notification was not sent intentionally,
+        and None if notification was attempted to be sent but an error occurred.
+        """
+        if (
+            self.registration.user
+            and self.registration.user.profile
+            and self.registration.user.profile.push_notifications
+        ):
+            # Do not send text if push_notifications is enabled
```

Comment:
I think we'll need some more checking eventually to handle the case where users log out of penn mobile on their phones, how do we re-enable texts? but this can be a conversation with mobile at some point.

Reply:
That's a good point; maybe we make enabling one automatically disable the other?

Reply:
Maybe if push notifications are enabled we grey out the phone number box on the web frontend with some sort of button to disable push / enable text

Reply:
I think it might be worth emphasizing on the sign-up screen if alerts are going to the app or via text. The push notification token should still be good even if they get logged out of the app, and if they delete the app I think the next time they sign-up for a class they would know to change it in most cases?

Reply:
Let's follow up on this with @ArmaanT and @r3khaan. Not sure how much of an issue it will be.